web/src/message_list_view.js

```diff
@@ -1347,7 +1347,17 @@ export class MessageListView {
         // prevent the appearance of tooltips whose reference has been removed.
         message_list_tooltips.destroy_all_message_list_tooltips();
         // Convert messages to list messages
-        let message_containers = messages.map((message) => this.message_containers.get(message.id));
+        let message_containers;
+        if (message_content_edited) {
+            // if the messages are edited, we want them locally echoed with new content.
+            message_containers = messages.map((message) => {
+                const message_container = this.message_containers.get(message.id);
+                message_container.msg.content = message.content;
+                return message_container;
```

Comment:
I think we need to be careful with mutating the core message object here; this may persist more than we'd expect.

Reply:
I think it won't be much of a problem. If the [`patch`](https://github.com/zulip/zulip/blob/c129b1779fc2d921784020b7e6a3c68bb7f8beaf/web/src/message_edit.js#L1050) request for `save_message_row_edit` fails, we can see that the original content is being restored, and is being re-rendered [here](https://github.com/zulip/zulip/blob/c129b1779fc2d921784020b7e6a3c68bb7f8beaf/web/src/message_edit.js#L1078), with [`message_content_edited`](https://github.com/zulip/zulip/blob/c129b1779fc2d921784020b7e6a3c68bb7f8beaf/web/src/echo.js#L275C11-L275C33) being set to false.

Manually testing by stopping the server and editing, it didn't seem like a problem either.